s4e-web/src/app/views/settings/settings.routes.ts

```diff
@@ -23,7 +20,12 @@ export const settingsRoutes: Routes = [
     canActivate: [IsLoggedIn],
     children: [
       {
-        path: 'dashboard',
+        matcher: adminDashboardMatcher,
+        component: AdminDashboardComponent,
+        canActivate: [IsAdminGuard]
+      },
+      {
+        matcher: managerDashboardMatcher,
         component: DashboardComponent,
         canActivate: [IsManagerGuard]
```

Comment:
I'm not sold on putting the admin/manager logic in between a matcher and a guard (see the other comment). If it's not possible to overload a path, I think the choice logic should be put in the routed component.

Reply:
I was thinking about it a lot. In your last comment, I wrote why the matcher is used. Now I will try to describe, why guard and matcher are implemented together.

1. Matcher chooses a page where you can land.
2. Guard secure route against unauthorized access.

Reply:
I disagree, matcher doesn't only choose the page, but also guards against someone unauthorized. I'd consider removing the guards, as they are redundant.

Reply:
I've done now double-check. I don't see any comparison between both solutions (matcher and guard), for the first look it seems reasonable to use only one of them at once. Let's hope that the missing guard doesn't have any nuances that will propagate some bugs or lack of security in the future.